### PR TITLE
Intern 4 (again)

### LINKED ISSUE #617: Update tests to Intern 4

### PR DESCRIPTION
**Pull request by bryanforbes**
**Type:** enhancement

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Rather than try to figure out the merge conflicts from master and Ed's PR, I did a clean conversion and cherry picked things from Ed's branch. One note: I did not indent the `tests: {}` blocks to make it easier to review. After approval and before merging, I will indent the `tests: {}` blocks.

Resolves #617

### CODE REVIEW

File: tests/functional/Drag.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
-import * as registerSuite from 'intern!object';
-import * as assert from 'intern/chai!assert';
+const { registerSuite } = intern.getInterface('object');
+const { assert } = intern.getPlugin('chai');
 // import pollUntil = require('intern/dojo/node!leadfoot/helpers/pollUntil');
```
Comment (agubler) [inline tests/functional/Drag.ts]: Can we remove this commented import?

File: tests/support/util.ts [hunk 2]
```diff
@@ -26,14 +26,14 @@ export function createResolvers() {
...
-			rAFStub.getCall(0).args[0]();
+			rAFStub.getCall(i).args[0]();
 		}
 		rAFStub.reset();
 	}
 
 	function resolveRIC() {
 		for (let i = 0; i < rICStub.callCount; i++) {
-			rICStub.getCall(0).args[0]();
+			rICStub.getCall(i).args[0]();
```
Comment (agubler) [inline tests/support/util.ts]: Thanks!

File: tests/unit/Registry.ts [hunk 3]
```diff
@@ -97,57 +96,50 @@ registerSuite({
...
-				const promise: any = new Promise((resolve) => {
-					resolveFunction = resolve;
+				const promise: Promise<any> = new Promise((resolve) => {
+					resolve(WidgetBase);
 				});
+
 				const factoryRegistry = new Registry();
 				factoryRegistry.define('my-widget', promise);
 				factoryRegistry.get('my-widget');
-				resolveFunction(WidgetBase);
```
Comment (agubler) [inline tests/unit/Registry.ts]: Isn’t the point of this test to resolve after the get?

Comment (bryanforbes) [reply tests/unit/Registry.ts]: True. I'll fix this and the next test.

**Pull request merged**